src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Double.cs

```diff
@@ -86,7 +87,13 @@ private void WriteNumberValueIndented(double value)
             JsonWriterHelper.WriteIndentation(output.Slice(BytesPending), indent);
             BytesPending += indent;
 
-            bool result = Utf8Formatter.TryFormat(value, output.Slice(BytesPending), out int bytesWritten);
+#if BUILDING_INBOX_LIBRARY
+            var format = new StandardFormat('G');
+#else
+            var format = new StandardFormat('G', 17);
+#endif
+
+            bool result = Utf8Formatter.TryFormat(value, output.Slice(BytesPending), out int bytesWritten, format);
```

Comment:
Just to be clear, we will never write +/- Infinity, etc, correct? But only allow parsing it (on the reader) if the payload happen to have a number that gets parsed as Infinity?

Reply:
Right, we can't write out `+/-Infinity` (and should have done the appropriate `ValidateDouble` checks before calling this method).

However, the IEEE spec says that values are parsed to their "infinitely precise" result and then rounded to the nearest representable result. This means that any valid number can be parsed but too large values round to `+/-Infinity` and to small values round to `+/-0`.

Reply:
why are we not just using G17 on .NET Core as well?

Reply:
We could, but it is overly verbose.

I also need to finish writing up a request to back port a fix to the utf8parser, as it currently disallows precision modifiers on G, for full framework (it's an artificial check and the underlying code already supports it, since we just forward to the Utf16 parser).